### PR TITLE
Finish moving query logic

### PR DESCRIPTION
**Pull request by drew-gross**
DatabaseController.prototype.find no longer depends on the mongo format, I think. Except maybe in count results.

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -653,34 +646,29 @@ DatabaseController.prototype.find = function(className, query, {
...
-          if (fieldName.match(/^authData\.([a-zA-Z0-9_]+)\.id$/)) {
-            throw new Parse.Error(Parse.Error.INVALID_KEY_NAME, `Cannot sort by ${fieldName}`);
-          }
-          const mongoKey = this.transform.transformKey(className, fieldName, schema);
-          mongoOptions.sort[mongoKey] = sort[fieldName];
-        }
+      // Parse.com treats queries on _created_at and _updated_at as if they were queries on createdAt and updatedAt,
+      // so duplicate that behaviour here. If both are specified, the corrent behaviour to match Parse.com is to
+      // use the one that appears first in the sort list.
+      if (sort && sort._created_at) {
```
Comment (peterdotjs) [inline src/Controllers/DatabaseController.js]: do we need the undefined check for `sort` since you've specified the default value?

Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: Probably not. Will change.

Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: Yep it was unnecessary, tests passed without.

**Comment by facebook-github-bot-1** [comment]
@drew-gross updated the pull request.

**Pull request merged**